### DISCUSSION TITLE
anti-virus says exe is a threat

### DISCUSSION THREAD

**Discussion by jazzjohn27** [pyinstaller/pyinstaller#5877] [PyInstaller]
Using the onefile option to create an exe file. Python 3.9.5, Win10.
When AVG scans the file, it reports threat Win64:Trojan-gen and quarantines the file.

 When I attempt to run the program, the file is removed and the error "Windows cannot access the specified device, path, or file. You may not have the appropriate permissions to access the item." appears.

These are the modules imported:
asyncio
re
kasa: Discover, SmartPlug
 time
 asyncio
subprocess
 psutil

Any ideas would be appreciated.

**Comment by rokm** [comment]
I guess about the only thing you can do is report your exe as false positive to the AV vendor.

Or avoid using `onefile` mode if you can.

**Reply by jazzjohn27** [reply]
How can I be sure it's a false positive and really not a threat? 
Is this common occurrence with Pyinstaller?

**Reply by bwoodsend** [reply]
It's a common occurrence with writing software in general. Python normally is ignored (which it really shouldn't be) because it's an interpreted language rather than a compiled one but as soon as you start writing EXEs then you're forever playing ping pong with overzelous AV..

**Reply by rokm** [reply]
> How can I be sure it's a false positive and really not a threat?

How can you be sure of anything I would tell you here?

You might try rebuilding the bootloader yourself, preferably with different toolchain that was used to build the prebuilt ones, to differentiate them a bit further. That is, assuming that you trust your environment (OS and the toolchain). Although I suspect in onefile mode, all collected python libraries increase similarity of the executable to other onefile-built executables to the extent that overeager pattern matchers might start to pick on it. 

> Is this common occurrence with Pyinstaller?

[Common enough](https://github.com/pyinstaller/pyinstaller/issues?q=is%3Aissue+virus) that we have a [separate label](https://github.com/pyinstaller/pyinstaller/issues?q=label%3Aantivirus-false-positives) for these kinds of reports now.

**Reply by jazzjohn27** [reply]
`How can you be sure of anything I would tell you here?`

 I ran AntiLiar and you came out as a truth teller...

Building the bootloader is beyond my skill set, so I'll try the single folder option.  Thanks much.

**Reply by eric15342335** [reply]
> Building the bootloader is beyond my skill set, so I'll try the single folder option. Thanks much.

```
git clone https://github.com/pyinstaller/pyinstaller
cd .\pyinstaller\bootloader
python .\waf all
cd ..\
pip install -e .
```
